File: 2020/otherLangs/cpp/01.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

int main(){
  ifstream inputfile;
  inputfile.open("./inputs/01.txt");
  vector<int> parsed = {};
  string line;
  while(getline(inputfile, line)){
    parsed.push_back(stoi(line));
  }
  int p1;
  int p2;
  for(int i = 0; i < parsed.size(); i++){
    for(int j = i + 1; j < parsed.size(); j++){
      if(parsed[i] + parsed[j] == 2020) p1 = parsed[i] * parsed[j];
      for(int k = j + 1; k < parsed.size(); k++){
        if(parsed[i] + parsed[j] + parsed[k] == 2020) p2 = parsed[i] * parsed[j] * parsed[k];
      }
    }
  }
  cout << "Part 1: " << p1 << endl;
  cout << "Part 2: " << p2 << endl;
}

File: 2020/otherLangs/golang/01.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {

	content, _ := ioutil.ReadFile("inputs/01.txt")
	text := string(content)
	s := strings.Split(text, "\n")
	nums := make([]int, len(s))
	var parsed int64
	var err error
	for i := 0; i < len(s); i++ {
		parsed, err = strconv.ParseInt(s[i], 10, 64)
		if err != nil {
			panic(err)
		}
		nums[i] = int(parsed)
	}
	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[i]+nums[j] == 2020 {
				fmt.Println("Part 1:")
				fmt.Println(nums[i] * nums[j])
			}
			for k := j + 1; k < len(nums); k++ {
				if nums[i]+nums[j]+nums[k] == 2020 {
					fmt.Println("Part 2:")
					fmt.Println(nums[i] * nums[j] * nums[k])
				}
			}
		}
	}
}
